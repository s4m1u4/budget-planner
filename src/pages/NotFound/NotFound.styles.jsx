export const box = {
  padding: "20px 0",
  display: "flex",
  flexDirection: "column",
  alignItems: "center",
};
