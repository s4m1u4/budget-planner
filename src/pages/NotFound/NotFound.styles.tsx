import styled from "@emotion/styled";

export const NotFoundBox = styled.div({
  padding: "20px 0",
  display: "flex",
  flexDirection: "column",
  alignItems: "center",
});
