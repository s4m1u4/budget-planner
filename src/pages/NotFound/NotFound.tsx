import React from "react";
import { Link } from "react-router-dom";
import { NotFoundBox } from "./NotFound.styles";
import pageNotFound from "../../assets/images/page-not-found.png";
import { Button, Container } from "@mui/material";

export const NotFound = () => {
  return (
    <Container>
      <NotFoundBox>
        <h1>Page not found</h1>
        <img src={pageNotFound} alt="Page not found" />
        <Button
          component={Link}
          to="/dashboard"
          variant="contained"
          color="success"
        >
          Come back
        </Button>
      </NotFoundBox>
    </Container>
  );
};
