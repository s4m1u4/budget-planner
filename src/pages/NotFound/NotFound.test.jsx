import { NotFound } from "./NotFound";
import { BrowserRouter } from "react-router-dom";
import { render, screen } from "@testing-library/react";
import pageNotFound from "../../assets/images/page-not-found.png";

const renderWithRouter = (Component) => {
  return render(<BrowserRouter>{Component}</BrowserRouter>);
};

describe("Not found component", () => {
  it("make snapshot", () => {
    const { baseElement } = renderWithRouter(<NotFound />);

    expect(baseElement).toMatchSnapshot();
  });

  describe("should render", () => {
    beforeEach(() => {
      renderWithRouter(<NotFound />);
    });

    it("should render a title", () => {
      expect(screen.getByText(/Page not found/i)).toBeInTheDocument();
    });

    it("should render an image", () => {
      const image = screen.getByRole("img");
      expect(image).toBeInTheDocument();
      expect(image).toHaveAttribute("alt", "Page not found");
      expect(image).toHaveAttribute("src", pageNotFound);
    });

    it('should render a button "Come back"', () => {
      expect(screen.getByText(/Come back/i)).toBeInTheDocument();
    });
  });
});
