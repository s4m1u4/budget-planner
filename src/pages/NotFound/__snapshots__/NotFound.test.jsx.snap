// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Not found component make snapshot 1`] = `
<body>
  <div>
    <div
      class="MuiContainer-root MuiContainer-maxWidthLg css-1oqqzyl-MuiContainer-root"
    >
      <div
        class="css-1v06gii"
      >
        <h1
          class="css-15poin4"
        >
          Page not found
        </h1>
        <img
          alt="Page not found"
          src="page-not-found.png"
        />
        <a
          class="MuiButton-root MuiButton-contained MuiButton-containedSuccess MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root btn css-sj6h5v-MuiButtonBase-root-MuiButton-root"
          href="/"
          tabindex="0"
        >
          Come back
          <span
            class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
          />
        </a>
      </div>
    </div>
  </div>
</body>
`;
